modelforge/curation/qm9_curation.py

```diff
@@ -373,16 +400,25 @@ def process(
             output_path=self.local_cache_dir,
             force_download=force_download,
         )
+        # clear out the data array before we process
+        self._clear_data()
```

Comment:
I am not complelty sure why this is necessary --- what is the motivation behind this call?

Reply:
The `self.data` list (which stores the processed records before being written to the hdf5 file)  is initialized in the constructor.  If someone were to initialize the class, and then call the `process()` function multiple types, it would just append the same records multiple times to `self.data`, if we didn't clear data in-between calls, which is just what `self._clear_data()` does.  

Since these aren't really functions that people are going to be using, really, directly at all, I didn't bother putting in any checking where if you call process a second time it would check the contents of data to see if only parts of it need to be reprocessed.   
wrapping data.clear() in it's own function was purely just for making improving the codecov metrics.